sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala

```diff
@@ -74,4 +79,30 @@ class HiveExplainSuite extends QueryTest {
       "Limit",
       "src")
   }
+
+  test("SPARK-6212: The EXPLAIN output of CTAS only shows the analyzed plan") {
+    withTempTable("jt") {
+      val rdd = sparkContext.parallelize((1 to 10).map(i => s"""{"a":$i, "b":"str$i"}"""))
+      read.json(rdd).registerTempTable("jt")
+      val outputs = sql(
+        s"""
+           |EXPLAIN EXTENDED
+           |CREATE TABLE t1
+           |AS
+           |SELECT * FROM jt
+      """.stripMargin).collect().map(_.mkString).mkString
+
+      val shouldContain =
```

Comment:
Can you also to check if it's real the optimized logical plan?
https://github.com/apache/spark/pull/7980/files#diff-dc57b297e91eedbf8cd67806640a6bbcR75

Free feel to take it. :)